packages/@aws-cdk/aws-vpcv2-alpha/lib/vpc-v2-base.ts

```diff
@@ -0,0 +1,368 @@
+import { Resource, Annotations } from 'aws-cdk-lib';
+import { IVpc, ISubnet, SubnetSelection, SelectedSubnets, EnableVpnGatewayOptions, VpnGateway, VpnConnectionType, CfnVPCGatewayAttachment, CfnVPNGatewayRoutePropagation, VpnConnectionOptions, VpnConnection, ClientVpnEndpointOptions, ClientVpnEndpoint, InterfaceVpcEndpointOptions, InterfaceVpcEndpoint, GatewayVpcEndpointOptions, GatewayVpcEndpoint, FlowLogOptions, FlowLog, FlowLogResourceType, SubnetType, SubnetFilter, CfnVPCCidrBlock } from 'aws-cdk-lib/aws-ec2';
+// eslint-disable-next-line no-duplicate-imports
+import { allRouteTableIds, flatten, subnetGroupNameFromConstructId } from '../lib/util';
+import { IDependable, Dependable, IConstruct } from 'constructs';
+
+/**
+ * Placeholder to see what extra props we might need,
+ * will be added to original IVPC
+ */
+export interface IVpcV2 extends IVpc {
+  /**
+   * The secondary CIDR blocks associated with the VPC.
+   *
+   * For more information, see the {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#configuring-secondary-cidr-blocks}.
+   */
+  readonly secondaryCidrBlock: CfnVPCCidrBlock[];
+
+  /**
+   * The primary IPv4 CIDR block associated with the VPC.
+   *
+   * For more information, see the {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#configuring-ipv4-cidr-block}.
+   */
+  readonly ipv4CidrBlock: string;
+}
+
+/**
+ * Base class for creating a VPC (Virtual Private Cloud) in AWS.
+ *
+ * For more information, see the {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html|AWS CDK Documentation on VPCs}.
+ */
+export abstract class VpcV2Base extends Resource implements IVpcV2 {
+
+  /**
+     * Identifier for this VPC
+     */
+  public abstract readonly vpcId: string;
+
+  /**
+     * Arn of this VPC
+     */
+  public abstract readonly vpcArn: string;
+
+  /**
+     * CIDR range for this VPC
+     */
+  public abstract readonly vpcCidrBlock: string;
+
+  /**
+     * List of public subnets in this VPC
+     */
+  public readonly publicSubnets: ISubnet[] = [];
+
+  /**
+     * List of private subnets in this VPC
+     */
+  public readonly privateSubnets: ISubnet[] = [];
+
+  /**
+     * List of isolated subnets in this VPC
+     */
+  public abstract readonly isolatedSubnets: ISubnet[];
+
+  /**
+     * AZs for this VPC
+     */
+  public readonly availabilityZones: string[] = [];
+
+  /**
+     * Identifier for the VPN gateway
+     */
+  //public readonly vpnGatewayId?: string;
+
+  /**
+     * Dependable that can be depended upon to force internet connectivity established on the VPC
+     */
+  public abstract readonly internetConnectivityEstablished: IDependable;
+
+  /**
+   * Newly added property for secondary CIDR blocks
+   */
+  public abstract readonly secondaryCidrBlock: CfnVPCCidrBlock[];
+
+  public abstract readonly ipv4CidrBlock: string;
```

Comment:
this is needed for ip validation in subnet, added docstring

Reply:
I believe we agreed to make it protected, do we need to add it in the Ivpc2 Interface, do you think it is needed out side the VPC construct?

Reply:
unfortunately an interface property cannot be protected, we need this in the interface since that is what we're passing here in the [subnet validating function](https://github.com/shikha372/aws-cdk/blob/vpc_full_control/packages/%40aws-cdk/aws-vpcv2-alpha/lib/subnet-v2.ts#L311) .

Reply:
and actually this will open another question, why these validation function are not a private functions in the VpcV2 class ?